lib/Tools/LinearLayout.cpp

```diff
@@ -655,19 +656,99 @@ LinearLayout::divideRight(const LinearLayout &divisor) {
                          divisor.getInDimSizeLog2(inDim));
   }
 
+  // Record size 1 out-dims caused by the division.
```

Comment:
I didn't review this part yet because I'm holding out hope we can keep the invariant that divideRight is the inverse of `*`.  :)